app/test/prepare_proposal_test.go

```diff
@@ -260,6 +267,122 @@ func TestPrepareProposalFiltering(t *testing.T) {
 	}
 }
 
+func TestPrepareProposalCappingNumberOfTransactions(t *testing.T) {
```

Comment:
[optional][not blocking] currently this test only create transactions with a single message. It may be worth adding a test case with transactions that contain multiple messages to verify that the number of messages is the limiting constraint, not the number of transactions. After we do that, we could also:

```suggestion
func TestPrepareProposalCappingNumberOfMessages(t *testing.T) {
```